src/test/java/org/apache/sysds/test/functions/builtin/BuiltinToOneHotTest.java

```diff
@@ -0,0 +1,77 @@
+package org.apache.sysds.test.functions.builtin;
+
+import org.apache.sysds.common.Types;
+import org.apache.sysds.lops.LopProperties;
+import org.apache.sysds.runtime.matrix.data.MatrixValue;
+import org.apache.sysds.test.AutomatedTestBase;
+import org.apache.sysds.test.TestConfiguration;
+import org.apache.sysds.test.TestUtils;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.HashMap;
+
+public class BuiltinToOneHotTest extends AutomatedTestBase {
+    private final static String TEST_NAME = "to_one_hot";
+    private final static String TEST_DIR = "functions/builtin/";
+    private static final String TEST_CLASS_DIR = TEST_DIR + BuiltinToOneHotTest.class.getSimpleName() + "/";
+
+    private final static double eps = 0;
+    private final static int rows = 10;
+    private final static int cols = 1;
+    private final static int numClasses = 10;
+
+    @Override
+    public void setUp() {
+        addTestConfiguration(TEST_NAME,new TestConfiguration(TEST_CLASS_DIR, TEST_NAME, new String[]{"B"}));
```

Comment:
I would say that one test case of a "nice" input is insufficient. If you want to see behavior, since i just added a Buildin function myself i suggest something in the direction of (Up for debate):
https://github.com/apache/systemml/blob/e80145b344068d68e53094e437524108d5f0f00a/src/test/java/org/apache/sysds/test/functions/builtin/BuiltinConfusionMatrixTest.java

Reply:
I added another test case that checks the stop condition if an incorrect number of classes is passed. The simple test case already runs on random input and verifies what the function does using a Java implementation in the test case. However, feel free to add more, if you deem it necessary.